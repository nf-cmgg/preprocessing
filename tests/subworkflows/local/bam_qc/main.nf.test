nextflow_workflow {

    name "Test Workflow BAM_QC"
    script "subworkflows/local/bam_qc/main.nf"
    workflow "BAM_QC"

    tag "subworkflows"
    tag "subworkflows/local"
    tag "subworkflows/local/bam_qc"

    test("Bam QC - HSmetrics") {

        when {
            workflow {
                """
                // [meta, bam, bai, roi, fasta, fai, dict]
                input[0] = Channel.of([
                    [ id:'test', single_end:false ],
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/cram/sample1.sorted.cram", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/cram/sample1.sorted.cram.crai", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/regions/roi_chr21.bed", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.dict", checkIfExists: true),
                ])
                // boolean
                input[1] = false
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.out.samtools_stats,
                workflow.out.samtools_flagstat,
                workflow.out.samtools_idxstats,
                file(workflow.out.picard_multiplemetrics[0][1][0]).name,
                file(workflow.out.picard_hsmetrics[0][1]).name,
            ).match()
        }

    }

    test("Bam QC - WGSmetrics") {
        when {
            workflow {
                """
                // [meta, bam, bai, roi, fasta, fai, dict]
                input[0] = Channel.of([
                    [ id:'test', single_end:false ],
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/cram/sample1.sorted.cram", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/cram/sample1.sorted.cram.crai", checkIfExists: true),
                    [],
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.dict", checkIfExists: true),
                ])
                // boolean
                input[1] = false
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.out.samtools_stats,
                workflow.out.samtools_flagstat,
                workflow.out.samtools_idxstats,
                file(workflow.out.picard_multiplemetrics[0][1][0]).name,
                file(workflow.out.picard_wgsmetrics[0][1]).name,
            ).match()
        }
    }

    test("Bam QC - Samtools") {
        when {
            workflow {
                """
                // [meta, bam, bai, roi, fasta, fai, dict]
                input[0] = Channel.of([
                    [ id:'test', single_end:false ],
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/cram/sample1.sorted.cram", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/cram/sample1.sorted.cram.crai", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/regions/roi_chr21.bed", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai", checkIfExists: true),
                    file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.dict", checkIfExists: true),
                ])
                // boolean
                input[1] = true
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.out.samtools_stats,
                workflow.out.samtools_flagstat,
                workflow.out.samtools_idxstats,
            ).match()
        }
    }
}
