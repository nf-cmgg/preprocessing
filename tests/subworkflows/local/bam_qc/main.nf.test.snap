{
    "Bam QC - HSmetrics": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,7098b9422bf3cce350116e93bd8cf83a"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,167e69b479663a15194ddf56cbc9e60e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,081d0431383fb7ea6b51b7077c6ec93c"
                ]
            ],
            "test.CollectMultipleMetrics.alignment_summary_metrics",
            "test.CollectHsMetrics.coverage_metrics"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T14:41:54.380478"
    },
    "Bam QC - Samtools": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,7098b9422bf3cce350116e93bd8cf83a"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,167e69b479663a15194ddf56cbc9e60e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,081d0431383fb7ea6b51b7077c6ec93c"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T14:43:20.206066"
    },
    "Bam QC - WGSmetrics": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,7098b9422bf3cce350116e93bd8cf83a"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,167e69b479663a15194ddf56cbc9e60e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,081d0431383fb7ea6b51b7077c6ec93c"
                ]
            ],
            "test.CollectMultipleMetrics.alignment_summary_metrics",
            "test.CollectWgsMetrics.coverage_metrics"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T14:42:57.86288"
    }
}